crates/filtering/src/hysteresis.rs

```diff
@@ -27,3 +27,29 @@ pub fn less_than_with_hysteresis(
                 -hysteresis
             }
 }
+
+pub fn greater_than_with_hysteresis_from_tresholds(
+    last_evaluation: bool,
+    value: f32,
+    lower_threshold: f32,
+    upper_threshold: f32,
```

Comment:
You may use `Range` here https://doc.rust-lang.org/std/ops/struct.Range.html

Reply:
I'm not sure if that really helps with readability, since we kinda want an inversed `Range`.

Reply:
I think this is exactly what we want to express. This is the "hysteresis-range" in that range the hysteresis is active

Reply:
Okay yes, but we don't need to know if it's inside the range but if its larger / smaller than the range. I don't know how to use a `Range` in this context, please suggest what you would like to see.

Reply:
You can just replace the `lower_threshold` and `upper_threshold` with a single `Range` or `RangeInclusive` with `start` and `end` replacing the `upper_threshold` and `lower_threshold`. The rest I'd probably keep very similar

Reply:
You mean like this:?

```
let upper = 2.0;
let lower = 1.0;
let last = false;

let value = greater_than_with_hysteresis_from_tresholds(last, 1.5, Range::from(lower..upper));


pub fn greater_than_with_hysteresis_from_tresholds(
    last_evaluation: bool,
    value: f32,
    hysteresis: Range<f32>,
) -> bool {
    if value > hysteresis.end {
        true
    } else if value < hysteresis.start {
        false
    } else {
        last_evaluation
    }
}
```

I don't really see the advantage.